platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationFilesPoller.java

```diff
@@ -102,67 +85,15 @@ public void register(@NotNull ChangeListener listener) {
 
     @Override
     public void run() {
-        try {
-            WatchKey key;
-
-            while (!Thread.currentThread()
-                    .isInterrupted()) {
-                key = watchService.take(); // blocking
-                LOGGER.debug("Key has been signalled.  Looping over events.");
-
-                for (WatchEvent<?> genericEvent : key.pollEvents()) {
-                    WatchEvent.Kind<?> kind = genericEvent.kind();
-
-                    if (kind == OVERFLOW || kind == ENTRY_MODIFY || kind == ENTRY_DELETE) {
-                        LOGGER.debug("Skipping event [{}]", kind);
-                        continue;
-                    }
-
-                    Path filename = (Path) genericEvent.context();
-
-                    if (!filename.toString()
-                            .endsWith(fileExtension)) {
-                        LOGGER.debug(
-                                "Skipping event for [{}] due to unsupported file extension of [{}].",
-                                filename,
-                                fileExtension);
-                        continue; // just skip to the next event
-                    }
-
-                    if (changeListener != null) {
-                        // Sleeping before notifying the listener to make sure file is
-                        // done writing, otherwise the listener may read the file too soon.
-                        TimeUnit.SECONDS.sleep(1);
-                        LOGGER.debug("Notifying [{}] of event [{}] for file [{}].",
-                                changeListener.getClass()
-                                        .getName(),
-                                kind,
-                                configurationDirectoryPath.resolve(filename));
-                        changeListener.notify(configurationDirectoryPath.resolve(filename));
-                    }
-                }
-
-                // Reset key, shutdown watcher if directory not able to be observed
-                // (possibly deleted)
-                if (!key.reset()) {
-                    LOGGER.warn("Configurations in [{}] are no longer able to be observed.",
-                            configurationDirectoryPath.toString());
-                    break;
-                }
-            }
-        } catch (InterruptedException | RuntimeException e) {
-            LOGGER.debug("The [{}] was interrupted.",
-                    this.getClass()
-                            .getName(),
-                    e);
-            Thread.currentThread()
-                    .interrupt();
+        while (!Thread.currentThread()
+                .isInterrupted()) {
+            watchService.checkAndNotify(); // blocking
```

Comment:
❓ Is this really a blocking call? It's not clear from the documentation. If not, we may want to have some kind of sleep between `checkAndNotify` calls to prevent a busy loop.